org-roam-db.el

```diff
@@ -234,6 +234,19 @@ If FILE is nil, clear the current buffer."
                       :where (= file $s1)]
                      file))
 
+;;;;; Protection against unique-ID errors
+(define-error 'emacsql-constraint "SQL constraint violation")
+(defun org-roam--protected-node-insert (query vec)
+  (catch 'sqlerror
+    (condition-case err
+        (org-roam-db-query query vec)
+      (emacsql-constraint
+       (throw 'sqlerror
+              (message "%s for %s (%s) in %s"
```

Comment:
Instead of message, let's use `lwarn` so it appears in a warning buffer. You can see examples of its usage elsewhere in the repo.

Reply:
`(describe-function 'lwarn)` ... cool! Will do.